python/ray/data/_internal/planner/map_batches.py

```diff
@@ -82,11 +83,15 @@ def process_next_batch(batch: DataBatch) -> Iterator[Block]:
                 else:
                     raise e from None
 
-            validate_batch(batch)
-            # Add output batch to output buffer.
-            output_buffer.add_batch(batch)
-            if output_buffer.has_next():
-                yield output_buffer.next()
+            if not isinstance(batch, GeneratorType):
+                batch = [batch]
+
+            for b in batch:
```

Comment:
should we wrap this for loop with the try-catch exception above at line 69-84?

Reply:
why? it wasn't wrapped before right?

Reply:
oh I was worried about that generator function not being executed, until the output is produced. So if users mutate the input batch in generator function, the exception is not caught as non-generator function.

Reply:
good point, updated